File: crates/dashboard-model/src/victron_controller/dashboard/from_0_1_0_decision.rs


pub fn convert__decision__from__0_1_0(from: &crate::victron_controller::dashboard::v0_1_0::decision::Decision) -> crate::victron_controller::dashboard::decision::Decision {
    crate::victron_controller::dashboard::decision::Decision {
        summary: from.summary.clone(),
        factors: serde_json::from_value(serde_json::to_value(&from.factors).unwrap()).unwrap(),
    }
}

File: web/src/model/victron_controller/dashboard/v0_2_0/from_0_1_0_decision.ts
// @ts-nocheck
import {Decision as v0_1_0_Decision} from '../v0_1_0/Decision'
import {Decision as v0_2_0_Decision} from './Decision'

export function convert__decision__from__0_1_0(from: v0_1_0_Decision): v0_2_0_Decision {
    return new v0_2_0_Decision (
        from.summary,
        JSON.parse(JSON.stringify(from.factors))
    )
}